src/software/new_geom/util/intersection.cpp

```diff
@@ -0,0 +1,182 @@
+#include "software/new_geom/util/intersection.h"
+
+/**
+ * Computes the point of intersection between two lines.
+ * Note: this computes the intersection of two lines, not line segments.
+ *
+ * See:
+ * https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection#Given_two_points_on_each_line
+ *
+ * @pre the lines are not parallel
+ *
+ * @params a, b points that represent the first line
+ * @params c, d points that represent the second line
+ *
+ * @return the point of intersection, if it exists
+ */
+std::optional<Point> intersection(const Point &a, const Point &b, const Point &c,
+                                  const Point &d)
+{
+    double x1 = a.x();
+    double y1 = a.y();
+    double x2 = b.x();
+    double y2 = b.y();
+    double x3 = c.x();
+    double y3 = c.y();
+    double x4 = d.x();
+    double y4 = d.y();
+
+    std::cout << "x1: " << x1 << std::endl;
+    std::cout << "y1: " << y1 << std::endl;
+    std::cout << "x2: " << x2 << std::endl;
+    std::cout << "y2: " << y2 << std::endl;
+    std::cout << "x3: " << x3 << std::endl;
+    std::cout << "y3: " << y3 << std::endl;
+    std::cout << "x4: " << x4 << std::endl;
+    std::cout << "y4: " << y4 << std::endl;
```

Comment:
Yeah left over from trying vertical line intersections. Will be removed when we sort out what to do with that

Reply:
I'll just leave my comment here, sorry for not getting around to an actual review yet. Why does the underlying implementation for `intersection(Line, Line)` have to be `intersection(Point, Point, Point, Point)`?

Reply:
I would also prefer not to add the `x` and `y` functions to `Line` since they aren't generally used. Could we implement`Line` intersection like this?
<img width="188" alt="image" src="https://user-images.githubusercontent.com/34253612/73909253-07510800-4861-11ea-9501-a529b4d81986.png">
where a<sub>1</sub>x + b<sub>1</sub>y + c<sub>1</sub>= 0  and a<sub>2</sub>x + b<sub>2</sub>y + c<sub>2</sub> = 0

Reply:
Thanks, yeah I'm currently looking into doing it that way instead